unyt/mpl_interface.py

```diff
@@ -81,6 +88,7 @@ def default_units(x, axis):
 
             Unit object
             """
+            axis._unytarrayname = x.name
```

Comment:
Can we avoid monkeypatching matplotlib objects? What if on the matplotlib side the `axis` object gets destroyed. We have no control over that so I think it would be better to figure out how to achieve the same thing fully inside of unyt.

Reply:
well, the Axis object passed to unyt_arrayConverter.axisinfo(unit, axis) does not contain or point to the underlying unyt_array for that axis, and the unit object can be just a string passed in from plt.plot(x, y, xunits="ms") for example.

I agree and my original implementation attempted to handle this solely within Unyt. In one approach, I added an ``owner`` attribute to Unit that would allow me to get back to the original unyt_array that has the ``name``, but matplotlib's design breaks this approach.

Reply:
I share @ngoldbaum's worry here.